test-requirements.in

```diff
@@ -4,13 +4,12 @@
 -r mypy-requirements.txt
 -r build-requirements.txt
 attrs>=18.0
-filelock>=3.3.0,<3.20.0  # latest version is not available on 3.9 that we still support
+filelock>=3.3.0
 lxml>=5.3.0; python_version<'3.15'
 psutil>=4.0
 pytest>=8.1.0
 pytest-xdist>=1.34.0
 pytest-cov>=2.10.0
 setuptools>=75.1.0
-tomli>=1.1.0  # needed even on py311+ so the self check passes with --python-version 3.9
+tomli>=1.1.0  # needed even on py311+ so the self check passes with --python-version 3.10
 pre_commit>=3.5.0
-platformdirs<4.5.0  # latest version is not available on 3.9 that we still support
```

Comment:
The pins for `filelock` and `platformdirs` were added in https://github.com/python/mypy/pull/20031. This change just reverse it now that Python 3.9 will be dropped.